src/mewbot/loader.py

```diff
@@ -103,11 +96,25 @@ def load_component(config: ConfigBlock) -> Component:
 
     target_class: Type[Any] = getattr(module, config["name"])
 
-    if not issubclass(target_class, Component):
+    # Verify that the implementation class matches the interface we got from
+    # the `kind:` hint.
+    if not issubclass(target_class, interface):
         raise Exception("Trying to load a non-registered class from config")
```

Comment:
This will be a theme with me when it comes to error messages. I'm a fan of maximal detail to give the user a debugging headstart. Though this does result in messier code.

Suggestion: Include a list of the currently registered classes - ideally broken down by type - so the user can easily see if they've just typod or if there is a more fundamental problem and the object they're trying to specify hasn't been loaded at all.